### PR TITLE
Random Key

### PR DESCRIPTION
**Pull request by piotrpalcz**
This PR is related to image-rs PR https://github.com/confidential-containers/guest-components/pull/385 and Shim/rune PR https://github.com/confidential-containers/enclave-cc/pull/256

Makes Runtime boot use key stored in file by image-rs/enclave agent.

After merging and fixes next step will be to seal/unseal the file with SGX

### CODE REVIEW

File: src/enclave-agent/Cargo.toml [hunk 1]
```diff
@@ -13,7 +13,7 @@ clap = "2.33.3"
 # logger module
 env_logger = "0.10.0"
 
-image-rs = { git = "https://github.com/confidential-containers/guest-components.git", default-features = false, rev = "fe8fb1a" }
+image-rs = { git = "https://github.com/piotrpalcz/image-rs.git", default-features = false, branch = "random_key_2" }
```
Comment (dcmiddle) [inline src/enclave-agent/Cargo.toml:16]: Just a note for later reference that we want to update this after https://github.com/confidential-containers/guest-components/pull/385 is merged

File: src/runtime-boot/init/src/main.rs [hunk 2]
```diff
@@ -4,30 +4,47 @@ extern crate serde_json;
...
+    nix::mount::mount(
+        Some(fs_type.as_str()),
+        mountpoint_c.as_c_str(),
+        Some(fs_type.as_str()),
+        flags,
+        Some("dir=/keys"),
+    )
+    .unwrap_or_else(|e| panic!("mount failed: {e}"));
+
+    let KEY_FILE: &str = "/mnt/key.txt";
```
Comment (dcmiddle) [inline src/runtime-boot/init/src/main.rs:44]: Double check this path matches the path in https://github.com/confidential-containers/guest-components/pull/385

**Review by dcmiddle — COMMENTED** [review]
Thanks for figuring out everything for this feature.
For commit messages I like the 7 rules here: https://cbea.ms/git-commit/ 
... add the why to the commit body and add a sign-off.

File: src/enclave-agent/Cargo.toml (hunk 1 above)
Comment (mythi) [reply src/enclave-agent/Cargo.toml:16]: to reflect that, I moved this PR to Draft.

File: src/runtime-boot/init/src/main.rs (hunk 2 above)
Comment (piotrpalcz) [reply src/runtime-boot/init/src/main.rs:44]: To boot/app enclave to run this need to match so if the e2e flow is working that means that the path is corresponding

**Comment by piotrpalcz** [comment]
> Thanks for figuring out everything for this feature. For commit messages I like the 7 rules here: https://cbea.ms/git-commit/ ... add the why to the commit body and add a sign-off.

Thanks, applied both of the suggestion - commit message is meaningful and signed-ff

**Review by dcmiddle — APPROVED** [review]

**Comment by piotrpalcz** [comment]
Closing, this PR, replacing with https://github.com/confidential-containers/enclave-cc/pull/312

**Pull request closed without merging**